web/layout.go

```diff
@@ -152,15 +152,21 @@ func (r *LayoutRender) addFileFromFS(templatesFS fs.FS, file string) {
 		"sum": func(a int, b int) int {
 			return a + b
 		},
-		"markdown": markdownToHTML,
-		"split":    strings.Split,
+		"markdown":      markdownToHTML,
+		"split":         strings.Split,
+		"includeBundle": includeBundle,
 	})
 	patterns := append([]string{r.root, file}, r.blocks...)
 	tmpl = template.Must(tmpl.ParseFS(templatesFS, patterns...))
 
 	r.addTemplate(name, tmpl)
 }
 
+func includeBundle(filename string) template.HTML {
```

Comment:
We don't really need this method.
We can add any specific js file in the template itself.
Check this as example:
https://github.com/trento-project/trento/blob/main/web/templates/clusters.html.tmpl#L2

We could this way simply change the template file to include the needed js script

Reply:
You're right, we can actually just include raw `script` tags but this allows us

- To do that in a much cleaner way
- To directly reference a webpack bundle

Other frameworks adopt similar techniques to reference static paths, also: https://docs.djangoproject.com/en/3.2/howto/static-files/

Reply:
I don't really get your point. I don't mean to include raw js code.
As far as I understand, here you only want to add the `<script src=\"/static/...>` element to the html directly (the link to the script).
We already do this in other parts of the code (include statis js files), like in the example link I put before.
As far as I see, you just implemented something we are already doing (and the current way looks better in my opinion).

Or am I missing something?

Reply:
Sorry, I meant raw `script` tags, not raw JS code :D

> We already do this in other parts of the code (include statis js files)

Yeah my point is that we shouldn't do that because if we modify the static assets directory in the future we will have to go manually through the whole project changing references to `/static/frontend/assets/js` :) that's why frameworks like Django or Phoenix have a `static_path` or `static` annotation for templates, that allows you to reference that path in a cleaner way. Also, I specifically created this for webpack bundles so we only have to worry about the actual filename, and not the whole path.

This allows us to streamline adding a new webpack bundle, just adding a new file, editing the webpack configuration and using this function in the template referencing the filename.

> As far as I see, you just implemented something we are already doing (and the current way looks better in my opinion).

IMHO it does not, but it's just my opinion :D

Reply:
Ok, now I understand better the motivation.
If you feel this is the way to go, just open other follow up ticket, as we should use the same pattern in all the places where we import static js files

Reply:
You're right, I have a plan for the other JS files :D

Reply:
Hadn't thought of it before, but I like this approach :-)

Reply:
To be honest, I agree with Xabi: I don't find this function particularly useful, since you are just moving a string interpolation from the actual template to a template function, but at the end of the day it's still just a path interpolated with a file name.

That said, if you want to keep it it's fine, but `includeBundle` doesn't seem exactly fitting, because you could include any kind of javascript file.
As a suggestion, consider calling it just `script()`, so that the end result in the template is `{{ script "check_results.js" }}`, which resembles what the original HTML was.

Reply:
As a side note, full-fledged web application frameworks have these kind of functions because they usually also have a super magical black boxes of complex asset management systems, which we fortunately don't. 😛

Reply:
Well I agree with you to some extent but what really Phoenix, Rails and Symfony expose under different names (like Symfony Encore) is just webpack with a sensible-defaults-oriented configuration, a thing that we fortunately do as well, even if we didn't rebrand it as "Trento Fireworks" :joy:

Anyway, ok I'll rename that to `script` if that looks reasonable to you :sweat_smile: